cola/main.py

```diff
@@ -0,0 +1,480 @@
+            'onto': args.onto,
+            'preserve_merges': args.preserve_merges,
+            'strategy': args.strategy,
+            'no_ff': args.no_ff,
+            'merge': args.merge,
+            'exec': getattr(args, 'exec', None), # python keyword
+            'keep_empty': args.keep_empty,
+            'force_rebase': args.force_rebase,
+            'strategy_option': args.strategy_option,
+            'stat': args.stat,
+            'no_stat': args.no_stat,
+            'verify': args.verify,
+            'rerere_autoupdate': args.rerere_autoupdate,
+            'root': args.root,
+            'autosquash': args.autosquash,
+            'committer_date_is_author_date': args.committer_date_is_author_date,
+            'ignore_date': args.ignore_date,
+            'whitespace': args.whitespace,
+            'ignore_whitespace': args.ignore_whitespace,
+            'C': args.context_lines,
+            'continue': getattr(args, 'continue', False), # python keyword
+            'abort': args.abort,
+            'skip': args.skip,
+            'edit_todo': args.edit_todo,
+            'upstream': args.upstream,
+            'branch': args.branch,
+            'capture_output': False,
+    }
+    status, out, err = cmds.do(cmds.Rebase, **kwargs)
+    if out:
+        core.stdout(out)
+    if err:
+        core.stderr(err)
+    return status
+
+
+def cmd_remote(args):
+    context = application_init(args)
+    from cola.widgets import editremotes
+    view = editremotes.new_remote_editor()
+    return application_start(context, view)
+
+
+def cmd_search(args):
+    context = application_init(args)
+    from cola.widgets.search import search
+    view = search()
+    return application_start(context, view)
+
+
+def cmd_stash(args):
+    context = application_init(args)
+    from cola.widgets.stash import stash
+    view = stash()
+    return application_start(context, view)
+
+
+def cmd_tag(args):
+    context = application_init(args)
+    from cola.widgets.createtag import new_create_tag
+    view = new_create_tag(name=args.name, ref=args.ref, sign=args.sign)
+    return application_start(context, view)
+
+
+# Windows shortcut launch features:
+
+def windows_find_git():
+    """Return the path of git.exe, or None if we can't find it."""
+    # For now, we assume that git is installed in one of the typical
+    # locations. This should be smarter.
+
+    pf = os.environ.get('ProgramFiles', 'C:\\Program Files')
+    pf32 = os.environ.get('ProgramFiles(x86)', 'C:\\Program Files (x86)')
+    for p in [pf32, pf]:
+        candidate = os.path.join(p, 'Git\\bin')
+        if os.path.isdir(candidate):
+            return os.path.join(candidate, 'git')
+
+def shortcut_launch():
+    """Launch from a shortcut
+
+    Prompt for the repository by default, and try to find git.
+    """
+    argv = ['cola', '--prompt']
+    git_path = windows_find_git()
+    if git_path is not None:
+        argv += ['--git-path', git_path]
+    main(argv)
```

Comment:
Pycharm. Fixed, and I've found the setting to add a newline by default, because that's been annoying me too.